docs/DeveloperGuide.adoc

```diff
@@ -297,69 +312,430 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |busy student |view schedule for the current week |prepare for them
+
+|`* * *` |busy student |view schedule for the upcoming week |prepare for them
+
+|`* * *` |new user |view all commands |learn how to use them
+
+|`* * *` |new user |view commands for a specific feature |learn how to use them
+
+|`* * *` |user |get reminded about how commands work |recall the commands
+
+|`* * *` |user |import and export data |easily migrate the data to another computer
+
+|`* *` |student |find a facilitator by name |locate details of facilitators without having to go through the entire list
+
+|`* *` |student |find tasks by date |keep track of tasks on a particular date
+
+|`* *` |student |find upcoming tasks |prioritise them
+
+|`* *` |busy student |find empty slots in my schedule |manage my time easily
 
-_{More to be added}_
+|`*` |student |mark a task as done |not take note of them anymore
+
+|`*` |student |add a priority level to a task |prioritise my tasks
+
+|`*` |student |tag my tasks |categorise them
+
+|`*` |student |see countdown timers |be reminded of deadlines
+
+|`*` |busy student |receive reminders about deadlines and events the next day |take note of them
+
+|`*` |student |mass delete the modules |delete them quickly once the semester is over
+
+|`*` |advanced user |use shorter versions of a command |type a command faster
+
+|`*` |careless user |undo my commands |undo the mistakes in my command
+
+|`*` |visual user |see a clear <<gui, GUI>> |navigate the App more easily
+|=======================================================================
 
 [appendix]
 == Use Cases
 
-(For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
+(For all use cases below, the *System* is the `Mod Manager` and the *Actor* is the `user`, unless specified otherwise)
+
+[discrete]
+=== Use case: UC01 - Add module
+
+*<<mss, MSS>>*
+
+1.  User requests to add a module.
+2.  User provides the module code and description of the module.
+3.  Mod Manager adds the module.
++
+Use case ends.
+
+*<<extensions, Extensions>>*
+
+[none]
+* 2a. Compulsory fields are not provided.
++
+[none]
+** 2a1. Mod Manager shows an error message.
+** 2a2. Mod Manager requests for the correct data.
++
+Use case resumes at step 2.
+
+* 2b. The module code is more than 10 characters or the description is more than 60 characters.
++
+[none]
+** 2b1. Mod Manager shows an error message.
+** 2b2. Mod Manager requests for the correct data.
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: UC02 - View module
+
+*<<mss, MSS>>*
+
+1.  User requests to view a module.
+2.  User provides the module code.
+3.  Mod Manager shows all information related to the module.
++
+Use case ends.
+
+*<<extensions, Extensions>>*
+
+[none]
+* 2a. The given module code is invalid.
++
+[none]
+** 2a1. Mod Manager shows an error message.
+** 2a2. Mod Manager requests for the correct data.
```

Comment:
What does requests for correct data mean? Does the Mod Manager show another error message?